pkg/restclient/common_test.go

```diff
@@ -36,3 +45,50 @@ func TestNewClientWithOpts_Error(t *testing.T) {
 	assert.NotNil(err)
 	assert.NotEqual(client.username, "testuser")
 }
+
+func TestWithTLSConfig(t *testing.T) {
+	assert := assert.New(t)
+	r := require.New(t)
+
+	certOut, err := ioutil.TempFile("", "cert")
+	r.Nil(err, "creating temp file")
+	defer os.Remove(certOut.Name())
+
+	cert, err := generateCert()
+	r.Nil(err, "generating dummy certificate")
+
+	err = pem.Encode(certOut, &pem.Block{Type: "CERTIFICATE", Bytes: cert})
+	r.Nil(err)
+
+	tlsOpts := &TLSOptions{
+		CaCertFile: certOut.Name(),
+	}
+
+	client, err := NewClientWithOpts(WithTLSConfig(tlsOpts))
+	r.Nil(err)
+
+	transport, ok := client.httpClient.Transport.(*http.Transport)
+	assert.True(ok)
```

Comment:
why we need to assert here, I think we can have `r.True(ok)`
instead of `assert.NotNil` you can use r.NotNil()